src/main/java/AaronBot.java

```diff
@@ -0,0 +1,275 @@
+            try {
+                indexCheck(taskIndexMark);
+            } catch (IndexErrorException e) {
+                System.out.println("Student, your mark index should be a number between 1 and "
+                        + taskList.size());
+                System.out.println(e);
+                return true;
+            }
+            if (markDone(Integer.parseInt(taskIndexMark))) {
+                System.out.println("Task has been marked successfully student \n" 
+                        + taskList.get(Integer.parseInt(taskIndexMark) - 1));
+            } else {
+                System.out.println("Task is already marked student! \n"
+                        + taskList.get(Integer.parseInt(taskIndexMark) - 1));
+            }
+            return true;
+        case "unmark":
+            String taskIndexUnmark = tokenizedUserInput[1];
+            try {
+                indexCheck(taskIndexUnmark);
+            } catch (IndexErrorException e) {
+                System.out.println("Student, your unmark index should be a number between 1 and "
+                        + taskList.size());
+                System.out.println(e);
+                return true;
+            }
+            if (unmarkDone(Integer.parseInt(taskIndexUnmark))) {
+                System.out.println("Task has been unmarked successfully student \n" 
+                        + taskList.get(Integer.parseInt(taskIndexUnmark) - 1));
+            } else {
+                System.out.println("Task is already unmarked student! \n"
+                        + taskList.get(Integer.parseInt(taskIndexUnmark) - 1));
+            }
+            return true;
+        case "show":
+            if (tokenizedUserInput.length > 1 && tokenizedUserInput[1].equals("list")) {
+                for (int i = 0; i < taskList.size(); i++) {
+                    System.out.println((i + 1) + ". " + taskList.get(i).toString());
+                }
+                return true;
+            } else {
+                throw new NonsenseCommandException(userInput);
+            }
+        case "delete":
+            String taskIndexDelete = tokenizedUserInput[1];
+            try {
+                indexCheck(taskIndexDelete);
+            } catch (IndexErrorException e) {
+                System.out.println("Student, your delete index should be a number between 1 and "
+                        + taskList.size());
+                System.out.println(e);
+                return true;
+        }
+        deleteTask(Integer.parseInt(taskIndexDelete));
+        return true;
+        case "bye":
+            System.out.println("Ok Student, HAND.");
+            try {
+                TaskListFile.writeToFile("TaskList.txt", taskList);
+            } catch (IOException e) {
+                System.out.println("Error saving file!\n" + e);
+            }
+            return false;
+        default:
+            throw new NonsenseCommandException(userInput);
+        }
+    }
```

Comment:
consider removing this comment as it does not improve the functionality of your code.